### PR TITLE
Create route for Book Suggestion

### PR DESCRIPTION
**Pull request by aleochoam**
### Summary
- Added the BookSuggestionsController
- Added the create action for the controller
- Skip the authentication action for the create action in the controller
- Set the user reference to optional

### Trello card
https://trello.com/c/VNswumKP/27-crear-ruta-para-creaci%C3%B3n-de-un-booksuggestion

### CODE REVIEW

File: app/controllers/api/v1/book_suggestions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module Api
+  module V1
+    class BookSuggestionsController < ApiController
+      skip_before_action :authenticate_api_v1_user!, only: :create
+
+      def create
+        @book_suggestion = BookSuggestion.new(book_suggestion_params)
```
Comment (Gmarn) [inline app/controllers/api/v1/book_suggestions_controller.rb]: Why did you use an `@` here? Seem useless, isn't it?

File: app/controllers/api/v1/book_suggestions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        @book_suggestion.user = current_api_v1_user
```
Comment (Gmarn) [inline app/controllers/api/v1/book_suggestions_controller.rb]: Maybe you can avoid this step if there is no current user.

File: app/controllers/api/v1/book_suggestions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+        if @book_suggestion.save
+          render json: @book_suggestion, status: :created
+        else
+          render json: { error: @book_suggestion.errors.messages }, status: :bad_request
+        end
+      rescue ActionController::ParameterMissing => e
```
Comment (Gmarn) [inline app/controllers/api/v1/book_suggestions_controller.rb]: Maybe you can define it more generaly this in your ApiController with a `rescue_from`.

File: app/controllers/api/v1/book_suggestions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        render json: { error: e.to_s }, status: :bad_request
+      end
+
+      private
+
+      def book_suggestion_params
+        params.require(:book_suggestion).permit(%i[title author link publisher year synopsis])
```
Comment (Gmarn) [inline app/controllers/api/v1/book_suggestions_controller.rb]: And the `price`?

File: app/controllers/api/v1/book_suggestions_controller.rb (hunk 1 above)
Comment (aleochoam) [reply app/controllers/api/v1/book_suggestions_controller.rb]: I have always done it that way, isn't it the way you treat the variable in the controller (same in RentController#create)?

Comment (Gmarn) [reply app/controllers/api/v1/book_suggestions_controller.rb]: Yes and not. If you define a variable with `@` you are creating a kind of global variable, this allow to access to this variable in all methods an other places like views. However here, you do not have a view, neither this variable is call in an other method. 
So you definition of `book_suggestion` as a global variable is useless.
Any doubts on that, send me a slack ;)

Comment (Gmarn) [reply app/controllers/api/v1/book_suggestions_controller.rb]: And for rent, you right, I missed it :)

**Pull request merged**